site/_posts/2023-03-09-react-example-application.md

```diff
@@ -0,0 +1,309 @@
+---
+layout: blog-post
+title: Using the FusionAuth React SDK in a React application
+description: In this guide, we'll showcase how to use the FusionAuth React SDK to implement the login, logout, and registration flows in a React application.
+author: Colin Frick
+category: blog
+image: blogs/react-example-application/react-example-application.png
+tags: tutorial tutorial-react tutorial-typescript react typescript oauth javascript
+excerpt_separator: "<!--more-->"
+---
+
+The React SDK from FusionAuth enables developers to implement the login, logout, and registration flows in a React application with ease.
+
+<!--more-->
+
+To show the React SDK in action, we will be using the FusionDesk application. FusionDesk is a simple help desk example application that allows users to create tickets and view them. The application is built using React and Node.js. It uses a server backend built using Ts.Ed, Node.js, and Typescript.
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-tickets.png" alt="FusionDesk showing the main list of tickets" class="img-fluid" figure=false %}
+
+To get started with your own React application, install the FusionAuth React SDK and follow the [FusionAuth React SDK](/docs/v1/tech/client-libraries/react-sdk) documentation.
+
+```bash
+npm install @fusionauth/react-sdk
+```
+
+### Prerequisites
+
+To follow along with this article, you will need to have the following installed:
+
+* Node.js
+* Git
+* Docker with Docker Compose (optional)
+
+{% include _callout-tip.liquid
+content=
+"If you want to run the example application on another domain than `localhost`, you need to make sure that FusionAuth runs on the same domain as the backend of the application. This is a limitation of the Hosted OAuth Service Provider Endpoints."
+%}
+
+{% include _callout-important.liquid
+content=
+"This article was written using an early look at the FusionAuth Hosted OAuth Service Provider Endpoints. The feature may be not yet available in the latest version of FusionAuth. You can track the progress at [Github fusionauth-issues#1943](https://github.com/FusionAuth/fusionauth-issues/issues/1943)"
+%}
+
+### Clone the Repository
+
+To begin, clone the demo application from GitHub using the following command:
+
+```bash
+git clone https://github.com/sonderformat-llc/fusionauth-example-react-fusiondesk
+```
+
+### Set up the FusionAuth Application
+
+To set up the FusionAuth application, follow the steps below:
+
+```bash
+cd fusionauth-example-react-fusiondesk
+docker compose up -d
+# Wait until FusionAuth is set up and available at http://localhost:9011
+cd server && cp example.env .env && npm install && npm run seed && cd ..
+cd client && cp example.env .env && npm install && cd ..
```

Comment:
I had some trouble with the 0.24.0 version of our React SDK, so I had to upgrade it:

```
diff --git a/client/package.json b/client/package.json
index d108474..bbd974b 100644
--- a/client/package.json
+++ b/client/package.json
@@ -6,7 +6,7 @@
     "@fortawesome/fontawesome-svg-core": "6.3.0",
     "@fortawesome/free-solid-svg-icons": "6.3.0",
     "@fortawesome/react-fontawesome": "0.2.0",
-    "@fusionauth/react-sdk": "0.24.0",
+    "@fusionauth/react-sdk": "^0.25.0",
     "@testing-library/jest-dom": "5.16.5",
     "@testing-library/react": "13.4.0",
     "@testing-library/user-event": "13.5.0",
```

Reply:
And I had to change [server/src/middlewares/fusion-auth.middleware.ts](https://github.com/sonderformat-llc/fusionauth-example-react-fusiondesk/blob/main/server/src/middlewares/fusion-auth.middleware.ts#L31) to reflect the change in the cookie name.

```diff
diff --git a/server/src/middlewares/fusion-auth.middleware.ts b/server/src/middlewares/fusion-auth.middleware.ts
index 3509b14..a55f84d 100644
--- a/server/src/middlewares/fusion-auth.middleware.ts
+++ b/server/src/middlewares/fusion-auth.middleware.ts
@@ -28,7 +28,7 @@ export class FusionAuthMiddleware implements MiddlewareMethods, OnInit {
     }
 
     async use(@Context() $ctx: Context, @Locals() locals: any) {
-        const {access_token} = $ctx.request.cookies;
+        const access_token = $ctx.request.cookies['app.at'];
 
         if (access_token) {
             try {
```